lib/puma/launcher.rb

```diff
@@ -298,7 +298,8 @@ def prune_bundler
 
       log '* Pruning Bundler environment'
       home = ENV['GEM_HOME']
-      bundle_gemfile = ENV['BUNDLE_GEMFILE']
+      bundle_gemfile = ENV['BUNDLER_ORIG_BUNDLE_GEMFILE']
+      bundle_gemfile = nil if bundle_gemfile == 'BUNDLER_ENVIRONMENT_PRESERVER_INTENTIONALLY_NIL'
```

Comment:
What's this about?

Reply:
Removed in 0b47735f78c370d103af5c1b8b581948f56cd688